components/script/dom/htmlmediaelement.rs

```diff
@@ -1350,6 +1368,35 @@ impl HTMLMediaElementMethods for HTMLMediaElement {
         self.paused.get()
     }
 
+    /// https://html.spec.whatwg.org/multipage/#dom-media-defaultplaybackrate
+    fn DefaultPlaybackRate(&self) -> Finite<f64> {
+        Finite::wrap(self.defaultPlaybackRate.get())
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#dom-media-defaultplaybackrate
+    fn SetDefaultPlaybackRate(&self, value: Finite<f64>) {
+        if *value != self.defaultPlaybackRate.get() {
+            self.defaultPlaybackRate.set(*value);
+            self.queue_ratechange_event();
+        }
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#dom-media-playbackrate
+    fn PlaybackRate(&self) -> Finite<f64> {
+        Finite::wrap(self.playbackRate.get())
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#dom-media-playbackrate
+    fn SetPlaybackRate(&self, value: Finite<f64>) {
+        if *value != self.playbackRate.get() {
+            self.playbackRate.set(*value);
+            if let Err(e) = self.player.set_rate(*value) {
+                eprintln!("Could not set the playback rate {:?}", e);
+            }
+            self.queue_ratechange_event();
```

Comment:
We receive the `Player::RateChanged` event from the media backend once the new rate has been actually set and we are already queueing the `ratechange` event in the `Player::RateChanged` handler, so there's no need to queue it here (we would get duplicated events).